capstone/src/main/java/com/google/step/servlets/PageViewServlet.java

```diff
@@ -0,0 +1,100 @@
+import java.io.IOException;
+
+import java.util.List;
+
+import com.google.step.clients.MetricsClient;
+import com.google.step.data.WeeklyPageView;
+import com.google.step.data.RestaurantPageViews;
+
+/**
+ * Servlet that will call the metricsClient for any doGet or doPost requests.
+ */
+@WebServlet("/page-view")
+public class PageViewServlet extends HttpServlet {
+
+    private static MetricsClient metricsClient = new MetricsClient();
+    private static Gson gson = new Gson();
+
+    /**
+     *
+     * @param request with a mix of the following parameters: <String, restaurantKey> and <int, year>.
+     *                These can be in the request or not at all, and depending on the mix a different
+     *                method from the client will be called.
+     *                String restaurantKey -> getCurrentPageViews()
+     *                int year, String restaurantKey -> getYearRestaurantPageViews()
+     *                empty -> getAllPageViews();
+     *                A SC_FORBIDDEN status will be set if none of these three specified combinations
+     *                is present.
+     * @param response Json String response that varies depending of the method called by the mix of
+     *                 parameters in the request.
+     *                 getCurrentPageViews() -> WeeklyPageView (as Json)
+     *                 getYearRestaurantPageViews() -> List<WeeklyPageView> (as Json list)
+     *                 getAllPageViews() -> List<RestaurantPageViews> (as Json list)
+     * @throws IOException can be thrown by the writer for the response.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String restaurantKey;
+        if (request.getParameter("restaurantKey") != null) {
+            restaurantKey = request.getParameter("restaurantKey");
+        } else {
+            restaurantKey = null;
+        }
+
+        int year = 0;
+        if (request.getParameter("year") != null) {
+            year = Integer.parseInt(request.getParameter("year"));
+        }
+
+        String json;
+        if (restaurantKey == null && year == 0) {
+            // getAllPageViews()
+            List<RestaurantPageViews> restaurantPageViewsList = metricsClient.getAllPageViews();
+            json = gson.toJson(restaurantPageViewsList);
+        } else if (year == 0 && restaurantKey != null) {
+            // getCurrentPageViews()
+            WeeklyPageView weeklyPageView = metricsClient.getCurrentPageViews(restaurantKey);
+            json = gson.toJson(weeklyPageView);
+
+        } else if (year != 0 && restaurantKey != null) {
+            // getYearRestaurantPageViews()
+            List<WeeklyPageView> weeklyPageViewList = metricsClient.getYearRestaurantPageViews(year, restaurantKey);
+            json = gson.toJson(weeklyPageViewList);
+
+        } else {
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
```

Comment:
Forbidden is more about authorization. This is probably more of an invalid argument (I imagine if they pass a year and no restaurant key, you end up in this state).

Reply:
Yeah that makes sense. Changed it to a SC_BAD_REQUEST.